### PR TITLE
V1.0.0

### PR DESCRIPTION
**Pull request by sHesl**

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -1,110 +1,920 @@
...
+      // document.cookie = 'cookie1=a';
+      // document.cookie = 'cookie2=b';
+      // document.cookie == 'cookie1=a;cookie2=b'; // true
+
+      // We mirror this behaviour via sinon stubs and an innerCookies property to maintain state
+      document.innerCookies = '';
+
+      sinon.stub(document, 'cookie').get(() => document.innerCookies);
+      sinon.stub(document, 'cookie').set((s) => {
+        document.innerCookies = !document.innerCookies ? s : document.innerCookies += '; ' + s;
```
Comment (icio) [inline test/test.js]:
```suggestion
        document.innerCookies = !document.innerCookies ? s : document.innerCookies + '; ' + s;
```

File: package.json [hunk 2]
```diff
@@ -1,41 +1,55 @@
...
+    "go-serve": "goexec 'http.ListenAndServe(\":8088\", http.FileServer(http.Dir(\".\")))'"
   },
   "repository": {
     "type": "git",
     "url": "git+https://github.com/unravelin/ravelinjs.git"
   },
-  "author": "Paul Scott <paul.scott@ravelin.com>",
-  "license": "Apache-2.0",
+  "author": "Ravelin (https://github.com/unravelin) <support@ravelin.com>",
+  "license": "TBD",
```
[the remark this answers is not in the document]
Comment (sHesl) [reply package.json]: Ya, just shows how old this PR is!

File: README.md [hunk 3]
```diff
@@ -59,10 +86,6 @@ and send that encrypted value (the cipher) back to your server.
 
 <script src="ravelin.min.js"></script>
 <script>
-    // Tracking.
-    ravelinjs.setPublicAPIKey('pk_live_...');
-    ravelinjs.trackPage();
-
```
[the remark this answers is not in the document]
Comment (sHesl) [reply README.md]: Was intending to show only the exact snippets of code needed for each example. You think it reads better how it was?

File: test/pages/scripttag/index.html [hunk 4]
```diff
@@ -7,73 +7,108 @@
...
-                    nameOnCard: document.getElementById('name').value,
-                    pan: document.getElementById('number').value,
-                    month: month.options[month.selectedIndex].value,
-                    year: document.getElementById('year').value,
-                });
-            });
-            return false;
-        };
-    </script>
+    <script>new TestPageSetup(ravelinjs);</script>
```
Comment (icio) [inline test/pages/scripttag/index.html]: This isn't an object.

```suggestion
    <script>TestPageSetup(ravelinjs);</script>
```

File: src/init.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+var RavelinJS = require('./core');
+
+// ========================================================================================================
+//
+// initialisation
+//
+
+RavelinJS.RavelinJS = RavelinJS;
```
[the remark this answers is not in the document]
Comment (sHesl) [reply src/init.js]: I'd rather RavelinJS wasn't exposed at all, so have taken this line out.

File: src/init.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+var rjs = new RavelinJS();
+
+// Register listeners
+if (typeof document !== 'undefined' && document.addEventListener) {
+  document.addEventListener('paste', onPaste);
+} else if (typeof document !== 'undefined' && document.attachEvent) {
+  document.attachEvent('paste', onPaste);
```
[the remark this answers is not in the document]
Comment (sHesl) [reply src/init.js]: One idea to improve this in [9f24ff](https://github.com/unravelin/ravelinjs/pull/19/commits/9f24ffc4dd98333942ca717ff62dfb1af7d639bd)

File: src/encryption.js [hunk 7]
```diff
@@ -0,0 +1,1044 @@
...
+  var encryptedHex = pubKey.encrypt(aesKeyB64 + '|' + ivB64);
+  var encryptedBits = sjcl.codec.hex.toBits(encryptedHex);
+  var aesKeyAndIVCiphertextBase64 = sjcl.codec.base64.fromBits(encryptedBits);
+
+  return aesKeyAndIVCiphertextBase64;
+}
+
+// ========================================================================================================
+//
+// vendored code
```
[the remark this answers is not in the document]
Comment (sHesl) [reply src/encryption.js]: I'd prefer we just kept the vendored stuff side-by-side (I have a vague memory of making a change to sjcl... something about error handling or alerts).

Moved it into it's own file though https://github.com/unravelin/ravelinjs/pull/19/commits/b22739a875275742954a92a483c7f3f302cb5ae2

**Comment by icio** [comment]
Things we should do when picking this back up:

* Replace all of the setters with an init() method which calls set* and track* calls by default.
* Update the deviceFinger
* Replace trackFingerprint() with identify().

**Pull request closed without merging**